um_run.py

```diff
@@ -0,0 +1,315 @@
+                ]
+            ),
+
+            # todo: ArchiveObjects(),
+
+            #
+            LinkExe(
+                # linker='gcc',
+                linker=os.path.expanduser('~/.conda/envs/sci-fab/bin/mpifort'),
+                flags=[
+                    '-lc', '-lgfortran', '-L', '~/.conda/envs/sci-fab/lib',
+                    '-L', os.path.expanduser('~/git/fab/tmp-workspace/gcom/build_output'), '-l', 'gcom'
+                ],
+                output_fpath='$output/../um_atmos.exe')
+        ],
+
+    )
+
+
+# TODO: REVIEWERS, SHOULD WE NEED THESE?
+# a not recommended flag?
+# todo: allow a list of filters?
+ALLOW_MISMATCH_FLAGS = [
+    ('*/hardware_topology_mod.f90', ['-fallow-argument-mismatch']),
+    ('*/setup_spectra_mod.f90', ['-fallow-argument-mismatch']),
+    ('*/mcica_mod.f90', ['-fallow-argument-mismatch']),
+    ('*/ios_comms.f90', ['-fallow-argument-mismatch']),
+    ('*/ios_client_queue.f90', ['-fallow-argument-mismatch']),
+    ('*/fastjx_specs.f90', ['-fallow-argument-mismatch']),
+    ('*/history_mod.f90', ['-fallow-argument-mismatch']),
+    ('*/lustre_control_mod.f90', ['-fallow-argument-mismatch']),
+    ('*/imbnd_hill_mod.f90', ['-fallow-argument-mismatch']),
+    ('*/io_configuration_mod.f90', ['-fallow-argument-mismatch']),
+    ('*/nlstcall_nc_namelist_mod.f90', ['-fallow-argument-mismatch']),
+    ('*/nlstcall_pp_namelist_mod.f90', ['-fallow-argument-mismatch']),
+    ('*/ios.f90', ['-fallow-argument-mismatch']),
+    ('*/regrid_alloc_calc_mod.f90', ['-fallow-argument-mismatch']),
+    ('*/halo_exchange_ddt_mod.f90', ['-fallow-argument-mismatch']),
+    ('*/halo_exchange_mpi_mod.f90', ['-fallow-argument-mismatch']),
+    ('*/halo_exchange_os_mod.f90', ['-fallow-argument-mismatch']),
+    ('*/mg_field_norm.f90', ['-fallow-argument-mismatch']),
+    ('*/rdbasis.f90', ['-fallow-argument-mismatch']),
+    ('*/io.f90', ['-fallow-argument-mismatch']),
+    ('*/ppxlook_mod.f90', ['-fallow-argument-mismatch']),
+    ('*/read_land_sea.f90', ['-fallow-argument-mismatch']),
+    ('*/diagopr.f90', ['-fallow-argument-mismatch']),
+    ('*/eg_bi_linear_h.f90', ['-fallow-argument-mismatch']),
+    ('*/glomap_clim_netcdf_io_mod.f90', ['-fallow-argument-mismatch']),
+    ('*/emiss_io_mod.f90', ['-fallow-argument-mismatch']),
+    ('*/ios_stash_server.f90', ['-fallow-argument-mismatch']),
+    ('*/io_server_listener.f90', ['-fallow-argument-mismatch']),
+    ('*/acumps.f90', ['-fallow-argument-mismatch']),
+    ('*/num_obs.f90', ['-fallow-argument-mismatch']),
+    ('*/io_server_writer.f90', ['-fallow-argument-mismatch']),
+    ('*/routedbl_mod.f90', ['-fallow-argument-mismatch']),
+    ('*/ios_init.f90', ['-fallow-argument-mismatch']),
+    ('*/eg_sl_helmholtz_inc.f90', ['-fallow-argument-mismatch']),
+    ('*/ukca_scenario_rcp_mod.f90', ['-fallow-argument-mismatch']),
+]
+
+
+def main():
+    logger = logging.getLogger('fab')
+    logger.setLevel(logging.DEBUG)
+    # logger.setLevel(logging.INFO)
```

Comment:
Tidy

Reply:
This stuff has changed in the latest configs. The logger defaults to info.